subsys/bluetooth/host/att.c

```diff
@@ -2959,6 +2999,69 @@ static int bt_att_accept(struct bt_conn *conn, struct bt_l2cap_chan **ch)
 BT_L2CAP_CHANNEL_DEFINE(att_fixed_chan, BT_L2CAP_CID_ATT, bt_att_accept, NULL);
 
 #if defined(CONFIG_BT_EATT)
+static k_timeout_t credit_based_connection_delay(struct bt_conn *conn)
+{
+	/*
+	 * 5.3 Vol 3, Part G, Section 5.4 L2CAP COLLISION MITIGATION
+	 * ... In this situation, the Central may retry
+	 * immediately but the Peripheral shall wait a minimum of 100 ms before retrying;
+	 * on LE connections, the Peripheral shall wait at least 2 *
+	 * (connPeripheralLatency + 1) * connInterval if that is longer.
+	 */
+
+	if (IS_ENABLED(CONFIG_BT_CENTRAL) && conn->role == BT_CONN_ROLE_CENTRAL) {
+		return K_NO_WAIT;
+
+	} else if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+		uint8_t random;
+		int err;
+
+		err = bt_rand(&random, sizeof(random));
+		if (err) {
+			random = 0;
+		}
+
+		const uint32_t rand_delay = random & 0x7; /* Small random delay for IOP */
+		const uint32_t calculated_delay =
+			2 * (conn->le.latency + 1) * BT_CONN_INTERVAL_TO_MS(conn->le.interval);
+
+		return K_MSEC(MAX(100, calculated_delay + rand_delay));
+	}
+
+	/* Must be either central or peripheral */
+	CODE_UNREACHABLE;
```

Comment:
Is there an advantage of using `CODE_UNREACHABLE` instead of ASSERT?

Reply:
I dont think it is possible to get here as the device is either central or peripheral. `CODE_UNREACHABLE` just silences a compiler warning.  `__ASSERT()` logs more information when hit in a quick test, so may be more useful.

Reply:
> I get a warning for unicast_audio_client sample without CODE_UNREACHABLE. Will add it back in again.

You can just put the ASSERT in a final `else { }` :)